### ISSUE TITLE
2 questions.

### ISSUE THREAD

**Issue by gitmko0** [blevesearch/bleve#1487]
message := struct{
	Id   string
	From string
	Body string
}

1. do you have a numeric 64bit / 128bit  for ID?
2. no way to revert back to using memory database instead of the default scorch? (is there a branch where we can test out memory / alternative db instead

**Comment by mschoch** [comment]
1.  Bleve has a numeric type that is designed to support numeric range queries (making a poor choice for IDs where you typically only perform exact matches).  Further, while the type encodes 64-bits of information, all of the higher-level code interprets the value as a float64 meaning you cannot represent all 2^64 integer values.  None of the bleve types support > 64-bits.

Overhauling Bleve's support for numeric values (most likely using something like BKD-tree as lucene does) is on the roadmap.

2.  Scorch is not the default in 1.x and we have not taken away the ability to use in-memory indexes.  That said, the index created by the `NewMemOnly` method is using some pretty old code.  The alternative that most people use is to invoke `NewUsing` and configure a KV-store of your choosing that doesn't persist to disk.

An in-memory index using scorch technology is available in the [bluge](https://github.com/blugelabs/bluge) developer preview, but it has not been back-ported to bleve.

**Comment by mschoch** [comment]
1.  It's difficult to say for sure.  The largest deployments I'm aware of are not owned by the companies I directly work with, but by their customers, so the information shared is limited.  I believe Couchbase does testing at 100 million items, but that is in aggregate split across a few separate bleve index partitions.

2.  Production ready means different things to different people.  There are already several companies I've worked with using Bleve in production.  Here are a few things you should think about:

- Bleve releases today are a best effort community release.  It means we do the best we can, and we think the releases are good.  But, this isn't enough for most of the companies I've worked with.  They all do their own testing internally before adopting a new release.
- You need to think about supporting indexes over time.  Over time bleve has evolved from the upsidedown index format to scorch.  And scorch has multiple file format versions within it.  Each of these came with significant benefits, but rebuilding indexes has a cost, so many users need to continue to support older indexes.
- You should think about how you plan to support your use of bleve.  Will you be digging into the code, maintaining a fork and proposing changes upstream?  Or will you prefer to pay someone to help you?  Your earlier questions suggest you don't plan on using scorch, and instead may use some other k/v store, and that may significantly limit the community support you receive as well as the paid options available to you.

3.  There are too many unknowns for me to be able to directly answer this.  The instances I'm aware of in the 100mil document range are all disk based, but you mentioned in-memory, so I have no experience to draw on there.  Because they are disk-based, the index itself is largely mmap'd meaning isn't part of the heap.  That said, Go runtime and linux kernel accounting change over time and there have been issues from time to time.  You probably should expect to face the usual challenges associated with such processes, tuning swap, over-commit, etc.  Finally, you offer some math about bytes per item, but again earlier you mentioned using the upsidedown index type with your own key/value store.  This uses an extremely wasteful encoding that most likely goes way past your budge.

In summary, I'd suggest you start with a smaller-scale test to see if the numbers are even close.  Also, if you go with a solution using the upsidedown index type and your own key/value store, there will be very little support.
